src/Microsoft.DotNet.Build.Tasks/CleanupVSTSAgent.cs

```diff
@@ -344,87 +325,73 @@ private DriveInfo ReportCommonDiskUsage(string dirType, string directory)
             return returnStatus;
         }
 
-        private async System.Threading.Tasks.Task<bool> CleanupDirectoryAsync(string directory, int attempts = 0)
+        private async System.Threading.Tasks.Task<bool> CleanupDirectoryAsync(string directory, int attempts = 0, bool ignoreExceptions = false)
         {
             try
             {
-                if (Directory.Exists(directory))
+                // A protected directory is, for instance, the directory which is currently running the cleanup task.
+                // The cleanup task should never try to clean itself up
+                foreach (string protectedDirectory in protectedDirectories)
+                {
+                    if (protectedDirectory.Contains(directory))
```

Comment:
I was surprised too.  I observed it occurring recently though.  Well, I observed that the cleanup agent task was running and then that something had tried to cleanup the directory it was running in and only the locked files which were part of this task remained (and the task was running long).  It's possible that something else was trying to clean this up at the same time but I thought it reasonable to attempt to safeguard against this occurrence.